infra.bs

```diff
@@ -280,6 +280,81 @@ quotes and monospace font.
 <a>ASCII lowercase</a> of <var>B</var>.
 <!-- TODO: define string equals? -->
 
+To <dfn data-export="">collect a sequence of characters</dfn> means that the following algorithm
+must be run, with <var>characters</var> being the set of characters that can be collected:</p>
+
+<ol>
+
+  <li><p>Let <var>input</var> and <var>position</var> be the same variables as
+  those of the same name in the algorithm that invoked these steps.</p></li>
+
+  <li><p>Let <var>result</var> be the empty string.</p></li>
+
+  <li><p>While <var>position</var> doesn't point past the end of <var>input</var> and the character
+  at <var>position</var> is one of the <var>characters</var>, append that character to the end of
+  <var>result</var> and advance <var>position</var> to the next character in
+  <var>input</var>.</p></li>
+
+  <li><p>Return <var>result</var>.</p></li>
+
+</ol>
+
+<p>The step <dfn>skip whitespace</dfn> means that the user agent must <a>collect a sequence of
+characters</a> that are <a>ASCII whitespace</a> characters. <!--The step
+<dfn>skip White_Space characters</dfn> means that the user agent must <a>collect a sequence of
+characters</a> that are <span>White_Space</span> characters. In both cases, the--> The collected
+characters are not used. <!--<ref spec=UNICODE>--></p>
+
+<p>When a user agent is to <dfn data-export="">strip line breaks</dfn> from a string, the user
+agent must remove any U+000A LINE FEED (LF) and U+000D CARRIAGE RETURN (CR) characters from that
+string.</p>
+
+<p>When a user agent is to <dfn data-export="">strip leading and trailing whitespace</dfn> from a
+string, the user agent must remove all <span data-x="space character">space characters</span> that
```

Comment:
But I'm not sure we need to define this one. It seems "remove any leading and trailing ASCII whitespace from _str_" is equally short.